db/version_set.cc

```diff
@@ -1187,6 +1187,11 @@ class LevelIterator final : public InternalIterator {
                   iend.Encode(), flevel_->files[i].smallest_key) < 0) {
             continue;
           }
+          auto const metadata = flevel_->files[i].file_metadata;
+          if (metadata->num_entries == metadata->num_range_deletions) {
```

Comment:
This check may have to be done in `SetFileIterator()` as well. I'm not sure how the test is passing though. Since `scan_opts_` is non-null, `SetFileIterator` would have called Prepare anyway.